File: leetcode/1344. 时钟指针的夹角.py
class Solution:
    def angleClock(self, hour: int, minutes: int) -> float:
        one_minute_angle = 6.0
        one_hour_angle = 30.0
        
        # 分针角度
        minute_angle = minutes * one_minute_angle
        
        # 时针角度：基础小时角度 + 分钟带来的偏移
        hour_angle = (hour % 12) * one_hour_angle + minutes * (one_hour_angle / 60.0)
        
        # 计算夹角差值的绝对值
        diff = abs(hour_angle - minute_angle)
        
        # 返回最小夹角
        return min(diff, 360.0 - diff)

File: leetcode/1344. 时钟指针的夹角.java
class Solution {
    public double angleClock(int hour, int minutes) {
        final double oneMinuteAngle = 6.0;
        final double oneHourAngle = 30.0;
        
        // 分针角度
        double minuteAngle = minutes * oneMinuteAngle;
        
        // 时针角度：基础小时角度 + 分钟带来的偏移
        double hourAngle = (hour % 12) * oneHourAngle + minutes * (oneHourAngle / 60.0);
        
        // 计算夹角差值的绝对值
        double diff = Math.abs(hourAngle - minuteAngle);
        
        // 返回最小夹角
        return Math.min(diff, 360.0 - diff);
    }
}